core.js

```diff
@@ -55,1407 +55,1421 @@ function _check(buffer, headers, options) {
 
 export async function fileTypeFromTokenizer(tokenizer) {
 	try {
-		return _fromTokenizer(tokenizer);
+		return new FileTypeParser().parse(tokenizer);
 	} catch (error) {
 		if (!(error instanceof strtok3.EndOfStreamError)) {
 			throw error;
 		}
 	}
 }
 
-async function _fromTokenizer(tokenizer) {
-	const buffer = Buffer.alloc(minimumBytes);
-	const bytesRead = 12;
-	const check = (header, options) => _check(buffer, header, options);
-	const checkString = (header, options) => check(stringToBytes(header), options);
-
-	// Keep reading until EOF if the file size is unknown.
-	if (tokenizer.fileInfo.size === undefined) {
-		tokenizer.fileInfo.size = Number.MAX_SAFE_INTEGER;
+class FileTypeParser {
+	check(header, options) {
+		return _check(this.buffer, header, options);
 	}
 
-	await tokenizer.peekBuffer(buffer, {length: bytesRead, mayBeLess: true});
+	checkString(header, options) {
+		return this.check(stringToBytes(header), options);
+	}
 
-	// -- 2-byte signatures --
+	async parse(tokenizer) {
+		this.buffer = Buffer.alloc(minimumBytes);
+		this.bytesRead = 12;
```

Comment:
Can you change this to not be on `this.` since it's only used in this scope?

Reply:
It's not only used in that scope, it's also by `this.check()`:

https://github.com/sindresorhus/file-type/blob/75a69b4d64a6f277c8ec628aed814caf33c2de2a/core.js#L67-L69

Reply:
This comment is attached to `this.bytesRead = 12;` and is what I am talking about. Not `this.buffer`.